getl/fileregistry/delta_diff.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from pyspark.sql import DataFrame, functions as F, types as T
+
+from ..block import BlockConfig
+from ..fileregistry import fileregistry_utils as fr_utils
+
+
+class DeltaDiff:
+    schema = T.StructType([T.StructField("version_date", T.TimestampType(), False)])
+
+    def __init__(self, bconf: BlockConfig) -> None:
+        self.file_registry_path = bconf.get("BasePath")
+
+        self.start_date = datetime.strptime(
+            bconf.get("DefaultStartDate"), "%Y-%m-%d %H:%M:%S"
+        )
+        self.current_date = datetime.utcnow()
+
+        self.join_on_fields = bconf.get("JoinOnFields")
+
+        self.spark = bconf.spark
+
+        dataframe = fr_utils.fetch_file_registry(self.file_registry_path, self.spark)
+        if dataframe:
+            self.start_date = dataframe.limit(1).collect()[0].version_date
```

Comment:
Will this always get the latest start_date? Should we add a failsafe of an order_by for example

Reply:
Don't think a failsafe is needed, the whole delta table will be overwritten after every lift